components/datadog/agent/install.go

```diff
@@ -109,6 +127,56 @@ func updateAgentConfig(
 	return lastCommand, pulumiAgentString, nil
 }
 
+func updateSystemProbeConfig(
+	fileManager *command.FileManager,
+	systemProbeConfig string,
+	os os.OS,
+	lastCommand *remote.Command) (*remote.Command, pulumi.StringInput, error) {
+
+	if systemProbeConfig == "" {
+		// don't update config
+		return lastCommand, nil, nil
```

Comment:
<!-- Warning highlight specific problems with the subject under review. These problems can be user-facing or behind the scenes and they are blockers. It is strongly recommended to pair this comment with a suggestion. If you are not sure if a problem exists or not, consider leaving a question. -->
**🔨 warning**
There is one particular situation where you might still want to update the config, being when you first write a config and then you want to clean it up coming back to the default empty config.  In other words, you probably want to write and empty config if you are called with `WithSystemProbeConfig('')`, while you don't want to touch the file if you are called without any mention of `WithSystemProbeConfig`

I suggest writing an empty file here, if it does no arm to have an empty config file always.

Reply:
cc @ogaca-dd as this was a recent bug you found on the agent config